kafka-client/src/main/java/dev/responsive/kafka/clients/ResponsiveRestoreConsumer.java

```diff
@@ -75,12 +106,64 @@ public void seek(final TopicPartition partition, final OffsetAndMetadata offsetA
 
   @Override
   public void seekToBeginning(final Collection<TopicPartition> partitions) {
-    final Collection<TopicPartition> withUncachedPosition = seekInitial(partitions);
+    final Collection<TopicPartition> withUncachedPosition = initializeOffsets(partitions);
     if (withUncachedPosition.size() > 0) {
       // Make sure to only do this if the list of partitions without cached start positions
       // is not empty. If it's empty, KafkaConsumer will seek to the actual beginning of _all_
       // partitions.
       super.seekToBeginning(withUncachedPosition);
     }
   }
+
+  @Override
+  public void pause(final Collection<TopicPartition> partitions) {
+    final Set<TopicPartition> toPause = new HashSet<>(pausedChangelogPartitions);
+    toPause.addAll(partitions);
+    super.pause(toPause);
+  }
+
+  @Override
+  public void resume(final Collection<TopicPartition> partitions) {
+    final Set<TopicPartition> toResume = new HashSet<>(partitions);
+    toResume.removeAll(pausedChangelogPartitions);
+    super.resume(toResume);
+  }
+
+  @Override
+  public void close() {
+    restoreListener.removeRestoreConsumer(this);
+    super.close();
+  }
+
+  @Override
+  public void close(final Duration timeout) {
+    restoreListener.removeRestoreConsumer(this);
+    super.close(timeout);
+  }
+
+  /**
+   * Mark this changelog partition as a (potential) standby to block restoration.
+   * <p>
+   * This method is invoked on all changelogs during changelog registration, which begins with a
+   * {@link #seek}. If the changelog is actually from an active task, it will be removed from this
+   * set and resumed immediately afterward when the changelog registration completes.
+   * See {@link #markChangelogAsActive(TopicPartition)} for more details
+   */
+  public void markChangelogAsStandby(final TopicPartition partition) {
```

Comment:
I can't find any usage of this method in this PR? also shouldn't this call `pause` ? (and shouldn't the below call `resume`?)